EndemolShineGroup/cosmiconfig-typescript-loader
# Travis to workflows

## What did you implement:



I came across [this issue](https://github.com/EndemolShineGroup/cosmiconfig-typescript-loader/pull/156) which is blocking my usage of craco which now depends on cosmiconfig. I read that the PR won't be accepted until the travis-ci porting to Github Actions was completed, so this serves as a foundation to complete that.

## How did you implement it:


I read [the original travis-ci config](https://github.com/EndemolShineGroup/cosmiconfig-typescript-loader/blob/develop/.travis.yml) and ported it to three workflows which separates the concerns:
- Build
  - Runs on supported LTS Node version, 12, 14, 16.
  - `build`
  - `lint`
  - `test`
- Coverage
  - Runs on LTS Node
  - `test`
- Release
  - Runs on LTS Node
  - `build`
  - `semantic-release`

## How can we verify it:



Check the workflow runs, these can be re-run until things work.

Secrets need to be added for:
- `SLACK_WEBHOOK` (if you plan on using it, otherwise can just remove this).
- `CC_TEST_REPORTER_ID` for code climate coverage.
- Release things:
  - `GITHUB_TOKEN`
  - `NPM_TOKEN`

Releasing will be tricky to test with it being automatic and run on master only, due to the nature of secrets and such, I can't assist here (but would if I could).

Let me know what you think, and what you would like to change, I'm happy to try and facilitate these changes where I can. 

## Tasks:



***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
